src/valarray.hpp

```diff
@@ -59,6 +59,7 @@ template <typename T, int d> class valarray
 };
 } // namespace quokka
 
+// array + array
```

Comment:
`valarray` is intended for representing mathematical vectors, with addition and multiplication by a scalar. I don't think it makes sense to use this object to add other operations. In general, it may be better to use a `std::array`.

Reply:
I'm trying to extend it to the capacity of std::valarray. Is there anything wrong with this? I know the C++ community doesn't use valarray much (e.g. https://twitter.com/blelbach/status/839743137761259521?lang=en ), but I don't really understand why. I find it very convenient. Is it a speed issue?

Reply:
I agree that it is very convenient. I'm not sure why the C++ community doesn't like it very much.

If you want to extend it, I am okay with it. Could you put the changes you want to make to quokka::valarray in a separate PR so I can review them independently of the multigroup code? That PR can be reviewed and merged before tackling the this PR with all of the multigroup stuff.